src/pages/resultsView/survival/SurvivalUtil.tsx

```diff
@@ -82,6 +120,103 @@ export function getEstimates(patientSurvivals: PatientSurvival[]): number[] {
     return estimates;
 }
 
+export function parseSurvivalData(s: string) {
+    const splitedStatusAndLabel = s.split(':').map(text => text.trim());
+    let status = undefined;
+    let label = undefined;
+    // survival data in new format "status:label"
+    if (splitedStatusAndLabel.length === 2) {
+        status = splitedStatusAndLabel[0];
+        label = splitedStatusAndLabel[1];
+    } else {
+        // old format "label"
+        label = splitedStatusAndLabel[0];
+    }
+
+    // if status not exists, status will be undefined
+    return {
+        status,
+        label,
+    } as ParsedSurvivalData;
+}
+
+export function getStatus(s: string, prefix: string) {
+    // if it's in predefined survival type
+    if (prefix in survivalClinicalDataVocabulary) {
+        return survivalClinicalDataVocabulary[prefix].includes(s);
+    }
+    // if it's custom survival type
+    else {
+        const parsedSurvivalData = parseSurvivalData(s);
+        if (parsedSurvivalData.status) {
+            return parsedSurvivalData.status === '1';
+        }
+    }
+    // return false as default for values that cannot get mapped
+    return false;
+}
+
+export function getSurvivalAttributePrefix(attributeIds: string[]) {
+    const attributesPrefix = _.reduce(
+        attributeIds,
+        (attributesPrefix, attributeId) => {
+            let prefix = attributeId.substring(
+                0,
+                attributeId.indexOf('_STATUS')
+            );
+            if (!attributesPrefix.includes(prefix)) {
+                if (attributeIds.includes(`${prefix}_MONTHS`)) {
+                    attributesPrefix.push(prefix);
+                }
+            }
+            return attributesPrefix;
+        },
+        [] as string[]
+    );
+    return attributesPrefix;
+}
+
+export function getSurvivalStatusAttributePrefix(attributeId: string) {
+    if (attributeId.indexOf('_STATUS')) {
```

Comment:
this would be a good application of regular expression (with capture groups) because you could accomplish this without seeking _STATUS twice.   "BLAH_STATUS".match(/([^_]*)_STATUS/)